src/g_game.c

```diff
@@ -766,7 +773,7 @@ void G_BuildTiccmd(ticcmd_t* cmd)
       // killough 10/98: make SG/SSG and Fist/Chainsaw
       // weapon toggles optional
       
-      if (!demo_compatibility && doom_weapon_toggles && !next_weapon)
+      if (!demo_compatibility && doom_weapon_toggles)
```

Comment:
I'm a bit torn on this. On the one hand, I'd find it logical that the prev/next buttons are not affected by the `doom_weapon_toggles` switch, on the other hand I'd wonder why this toggle had an effect on weapon switching in one way but not the other.